Assets/MixedRealityToolkit.Examples/Demos/UX/Collections/Scripts/GridObjectLayoutControl.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UnityEngine.UI;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    /// <summary>
+    /// Provides functions to control layout of GridObjectCollection as well
+    /// as to output positions of child controls to help with building
+    /// GridObjectCollectionTests.
+    /// </summary>
+    public class GridObjectLayoutControl : MonoBehaviour
+    {
+        [Tooltip("Point this at the GridObjectCollection to control.")]
+        public GridObjectCollection grid;
+        [Tooltip("Optional text field to output the layout of control.")]
+        public TMPro.TextMeshPro text;
+
+        /// <summary>
+        /// Change the grid collection's layout to the next one in order.
+        /// </summary>
+        public void NextLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) + 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Change the grid's layout to the previous one in order
+        /// </summary>
+        public void PreviousLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) - 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Use this to run mock tests in grid object collection tests
+        /// and print the resulting positions of child objects to a file.
```

Comment:
Nit: It could be helpful for a reader to know the rough template/name of the file to look for (i..e look for a file with the name printgrid-YYYYMMDD.txt in XYZ folder). Otherwise they will need to look into the implementation details (which I guess in this case, also isn't that bad).

Reply:
Ah actually I guess it's printed out, so maybe no biggie.